pkg/apis/sources/v1alpha1/apiserver_types.go

```diff
@@ -67,6 +67,11 @@ type ApiServerSourceSpec struct {
 	// Sink is a reference to an object that will resolve to a domain name to use as the sink.
 	// +optional
 	Sink *corev1.ObjectReference `json:"sink,omitempty"`
+
+	// Mode is the mode the receive adapter controller runs under: Ref or Resource.
+	// `Ref` sends only the reference to the resource.
+	// `Resource` send the full resource.
+	Mode string `json:"mode,omitempty"`
```

Comment:
Shouldn't this be validated somewhere?

Reply:
it defaults to ref if set to a random thing, see the Trash test.

Reply:
Yes, but it would be better if we could validate it when the resource is created.

Reply:
Ah got it. How about we do a followup to wire in the event sources into the webhook and implement the validatable stuff?

Reply:
https://github.com/knative/eventing/issues/1196